src/garage/_dtypes.py

```diff
@@ -474,6 +474,146 @@ class StepType(enum.IntEnum):
     TIMEOUT = 3
 
 
+class EnvStep(
```

Comment:
can you put this in `_environment.py` instead? It is only used by the garage.Environment interface, not the rest of garage.